wicket-extensions/src/main/java/org/apache/wicket/extensions/ajax/markup/html/AjaxLazyLoadPanel.java

```diff
@@ -139,56 +182,176 @@ protected void handleCallbackScript(final IHeaderResponse response,
 	@Override
 	protected void onBeforeRender()
 	{
-		if (state == 0)
+		if (state == LoadingState.WAITING_FOR_LAZY_COMPONENT)
 		{
 			add(getLoadingComponent(LAZY_LOAD_COMPONENT_ID));
-			setState((byte)1);
+			setState(LoadingState.REPLACING_LAZY_COMPONENT);
 		}
 		super.onBeforeRender();
 	}
 
+	@Override
+	public void onEvent(IEvent<?> event)
+	{
+		super.onEvent(event);
+
+		if (state == LoadingState.REPLACING_LAZY_COMPONENT)
```

Comment:
No, as `isReadyForReplacement()` is actually there for preventing replacement prematurely. Why would you wait until some time in the future to do the replacement when you can replace now and the component is ready for replacement?